proposed/http-message.md

```diff
@@ -1133,8 +1141,8 @@ interface StreamableInterface
      * @param int $length Read up to $length bytes from the object and return
      *     them. Fewer than $length bytes may be returned if underlying stream
      *     call returns fewer bytes.
-     * @return string|false Returns the data read from the stream, false if
-     *     unable to read or if an error occurs.
+     * @return string Returns the data read from the stream
+     * @throws \RuntimeException if unable to read or if an error occurs.
```

Comment:
What does "unable to read" mean? Throw an exception when a non blocking stream has no data? Also, what would `''` mean? Does it signify EOF if you requested more than 0 bytes?

Here's how Python describes their read() method: https://docs.python.org/3.1/library/io.html#io.RawIOBase.read. The takeaway would be:

> Fewer than n bytes may be returned if the operating system call returns fewer than n bytes.
> If 0 bytes are returned, and n was not 0, this indicates end of file. If the object is in non-blocking mode and no bytes > are available, None is returned.

Maybe we need something along those lines to be more explicit.